lua/neotest-dart/init.lua

```diff
@@ -138,35 +138,46 @@ function adapter.build_spec(args)
   if not tree then
     return {}
   end
+
   local position = tree:data()
 
   local command_parts = {}
 
   if position.type == 'dir' then
-    command_parts = {
-      command,
-      'test',
-      '--reporter',
-      'json',
-    }
+      if string.sub(position.path, - #'/test') == '/test' then
+          command_parts = {
+              command,
+              'test',
+              position.path,
+              '--reporter',
+              'json',
+          }
+      else
+          command_parts = {
+              command,
+              'test',
+              string.format("%s/%s", position.path, 'test'),
+              '--reporter',
+              'json',
+          }
+      end
   end
 
-  if position.type == 'test' or position.type == 'file' then
-    local test_argument = construct_test_argument(position, args.strategy)
```

Comment:
I see this parameter is used later in 161(171) line. Should it be deleted?

Reply:
My mistake. Thx!